### PR TITLE
[cloud_firestore] add in and array-contains-any

### LINKED ISSUE #1333: [cloud_firestore] implement array-contains-any
Do you plan to implement the "array-contains-any" feature?

https://dev.to/nabettu/array-contains-any-was-created-in-firebase-firestore-so-i-will-explain-it-with-a-sample-site-4g9

### LINKED ISSUE #1387: [cloud_firestore] Support `in` queries
[`in` queries were recently added to Cloud Firestore](https://firebase.googleblog.com/2019/11/cloud-firestore-now-supports-in-queries.html), i.e. the native client libraries support this feature and the Flutter plugin should support it as well.

### PR DESCRIPTION
**Pull request by rosariopfernandes**
## Description

Firestore now supports `in` and `array-contains-any` queries. This PR should add those queries to FlutterFire.

## Related Issues

Close #1333 
Close #1387 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by kroikie** [comment]
@rosariopfernandes Thanks very much for this, would you mind updating the tests so that this functionality is included?

**Comment by rosariopfernandes** [comment]
@kroikie I've added unit and integration tests. Please have a look

File: packages/cloud_firestore/lib/src/query.dart [hunk 1]
```diff
@@ -126,6 +126,8 @@ class Query {
     dynamic isGreaterThan,
     dynamic isGreaterThanOrEqualTo,
     dynamic arrayContains,
+    dynamic arrayContainsAny,
+    dynamic whereIn,
```
Comment (slightfoot) [inline packages/cloud_firestore/lib/src/query.dart]: arrayContainsAny and whereIn should be `List<dynamic>`

Comment (rosariopfernandes) [reply packages/cloud_firestore/lib/src/query.dart]: @slightfoot Thanks! Changed it.

**Comment by awazgyawali** [comment]
What's the progress going on here? I would love to contribute if needed.

**Comment by slightfoot** [comment]
@kroikie @collinjackson this is an important feature to get into the plugin ASAP. Its been requested for years from the Firebase team and finally it here' lots of developers want to use these new features.

Now the tests are in place think we can get this reviewed and merged?

**Comment by awazgyawali** [comment]
@rosariopfernandes Is it possible to add me as a contributor in your fork.

**Comment by collinjackson** [comment]
I will review this soon. Thank you!

**Comment by awazgyawali** [comment]
@googlebot I consent.

**Comment by georgeherby** [comment]
Is there any reason that we are using this PR as opposed to the one by @awazgyawali (#1477 ) which has been closed. 1477 is only one commit behind master and look pretty clean, this, however, has its fair share of conflicts and is quite behind master?

I agree with @slightfoot that this is a really important feature and much needed, so seems silly to make our lives more difficult?

**Comment by collinjackson** [comment]
I've re-opened #1477 per @georgeherby's comment. Let's see if the tests pass and maybe I can land that one.

**Comment by awazgyawali** [comment]
I am happy if either of the PR is merged, it's just hard for me to manage this PR via another PR. Let @rosariopfernandes decide on this.

**Comment by rosariopfernandes** [comment]
I'm also happy if either of the PRs is merged.

**Pull request closed without merging**